hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -161,7 +162,17 @@ private[hudi] object HoodieSparkSqlWriter {
       // Convert to RDD[HoodieKey]
       val keyGenerator = DataSourceUtils.createKeyGenerator(toProperties(parameters))
       val genericRecords: RDD[GenericRecord] = AvroConversionUtils.createRdd(df, structName, nameSpace)
-      val hoodieKeysToDelete = genericRecords.map(gr => keyGenerator.getKey(gr)).toJavaRDD()
+      val hoodieKeysToDelete = if (HoodieIndex.GLOBAL_INDICES.contains(parameters(HoodieIndexConfig.INDEX_TYPE_PROP))) {
```

Comment:
There is a NonPartitionedKeyGenerator that you can pass and use for deletions. Since as a client you are aware of 2 facts a) Either you are using GLOBAL_INDEX or not b) If this batch is for Deletions, you can configure the hudi client accordingly with a correct partition generator -> https://github.com/apache/incubator-hudi/blob/master/hudi-spark/src/main/java/org/apache/hudi/NonpartitionedKeyGenerator.java

Reply:
@n3nash thanks for bringing that keyGenerator to my attention. That better serves my use case. Do you think extending it into a "GlobalDeleteKeyGenerator" would be a valid change instead?

Reply:
I think just create an entirely new keyGenerator since tables with GlobalIndex can still be partitioned, all we are doing is here allowing us to pass just the record_keys to be able to delete those records.